src/app_service/service/wallet_account/service.nim

```diff
@@ -821,22 +821,31 @@ QtObject:
       return
     return keycardsWithKeycardUid[0]
 
-  proc getAllKnownKeycardsGroupedByKeyUid*(self: Service): seq[KeycardDto] = 
+  proc getAllKnownKeycardsGroupedByKeyUid*(self: Service): seq[KeycardDto] =
     try:
       let response = backend.getAllKnownKeycardsGroupedByKeyUid()
       if responseHasNoErrors("getAllKnownKeycardsGroupedByKeyUid", response):
         return map(response.result.getElems(), proc(x: JsonNode): KeycardDto = toKeycardDto(x))
     except Exception as e:
       error "error: ", procName="getAllKnownKeycardsGroupedByKeyUid", errName = e.name, errDesription = e.msg
 
-  proc getKeycardByKeyUid*(self: Service, keyUid: string): seq[KeycardDto] = 
+  proc getKeycardByKeyUid*(self: Service, keyUid: string): seq[KeycardDto] =
     try:
       let response = backend.getKeycardByKeyUid(keyUid)
       if responseHasNoErrors("getKeycardByKeyUid", response):
         return map(response.result.getElems(), proc(x: JsonNode): KeycardDto = toKeycardDto(x))
     except Exception as e:
       error "error: ", procName="getKeycardByKeyUid", errName = e.name, errDesription = e.msg
 
+  proc isKeycardAccount*(self: Service, account: WalletAccountDto): bool =
+    if account.isNil or
+      account.keyUid.len == 0 or
+      account.path.len == 0 or
+      utils.isPathOutOfTheDefaultStatusDerivationTree(account.path):
+        return false
+    let keycards = self.getKeycardByKeyUid(account.keyUid)
```

Comment:
Well that's a question we discussed, do we want to call status-go whenever we need something or map status-go state on the desktop app side. Then we may raise a question how much are expensive those calls and so on.

Reply:
For sure not a must have but really a nice to have.
Status-go call are expensive, IMO it is better to have in memory state
And yes in that case we need to update cache on change

Reply:
@alaibe ok, will update my changes here and store it on the client side.

Reply:
@alaibe I've checked this once again, that change, mapping all keycards' related data in a cache on the client side and maintaining it in parallel with the status-go is a work by itself, cause it requires a bit of refactor, so I guess it would be better to do that in a separate PR.